File: 2019 PAC II - AED - 0800 - Semana de Ejercicios - Ejercicio 1/Python/Caja.py
from Lista import*

class Caja:
    
    def __init__(self):
        self.primerGrupo = None
    
    def agregarFila(self):
        
        if(not self.primerGrupo):
            self.primerGrupo = Nodo(ListaEnlazada())
            return True
        
        else:
            pila = self.primerGrupo
            self.primerGrupo = Nodo(ListaEnlazada())
            self.primerGrupo.siguiente = pila
            return True
        
        return False

    def agregarObjeto(self,valor, posicion = None):
        return self.primerGrupo.valor.agregar(valor, posicion)
    
    def imprimirCaja(self):
        camino = ""
        actual = self.primerGrupo
    
        while(actual.siguiente):
            camino = ("%s%s -> " %(camino, actual.valor))
            actual = actual.siguiente
        
        camino = ("%s%s -> None" %(camino, actual.valor))
    
        print(camino)
    




    

File: 2019 PAC II - AED - 0800 - Semana de Ejercicios - Ejercicio 1/JavaScript/Caja.js
function Caja(){
    //atributos
    this.primerGrupo = null;

    //metodos
    this.agregarFila = CajaAgregarFila;
    this.agregarObjeto = CajaAgregarObjeto;
    //this.imprimir = CajaImprimir;
}

    function CajaAgregarFila(){
        if(!this.primerGrupo){
            this.primerGrupo = new Nodo(new ListaEnlazada());
            return true;
        }
        else{
            pila = this.primerGrupo;
            this.primerGrupo = new Nodo(new ListaEnlazada())
            this.primerGrupo.siguiente = pila;
            return true;
        }
        return false;
    }

    function CajaAgregarObjeto(valor, posicion = null){
        return this.primerGrupo.valor.agregar(valor, posicion);
    }

    /*
    Función Experimental aun sin corregir

    function CajaImprimir(){
        actual = this.primerGrupo;

        while(actual.siguiente){
            actual.valor.imprimirLista();
            actual = actual.siguiente;
        }
        actual.valor.imprimirLista();
    }
     */